ios/Classes/PTPluginUtils.m

```diff
@@ -20,26 +58,385 @@ + (NSNumber *)PT_idAsNSNumber:(id)value
+    }
+    
+    return resultAnnot;
+}
+
++ (NSString *)PT_idToJSONString:(id)infoId {
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:infoId options:0 error:nil];
+    return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+}
+
++ (id)PT_JSONStringToId:(NSString *)jsonString {
+    NSData *annotListData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+    return [NSJSONSerialization JSONObjectWithData:annotListData options:kNilOptions error:nil];
+}
+
+#pragma mark - Functions
+
++ (void)importAnnotations:(NSString *)xfdfCommand resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    NSError* error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if([doc HasDownloader])
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
+        
+        PTFDFDoc *fdfDoc = [PTFDFDoc CreateFromXFDF:xfdfCommand];
+        
+        [doc FDFUpdate:fdfDoc];
+        [docVC.pdfViewCtrl Update:YES];
+        
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: There was an error while trying to import annotations. %@", error.localizedDescription);
+    }
+    
+    flutterResult(nil);
+}
+
++ (void)exportAnnotations:(NSString *)annotationList resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    NSError* error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if([doc HasDownloader])
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
+        
+        if (!annotationList) {
+            PTFDFDoc *fdfDoc = [doc FDFExtract:e_ptboth];
+            flutterResult([fdfDoc SaveAsXFDFToString]);
+        } else {
+            PTVectorAnnot *annots = [[PTVectorAnnot alloc] init];
+            
+            
+            NSArray *array = [PTPluginUtils PT_idAsArray:[PTPluginUtils PT_JSONStringToId:annotationList]];
+            
+            for (NSDictionary *annotation in array) {
+                NSString *annotationId = annotation[PTAnnotIdKey];
+                int pageNumber = ((NSNumber *)annotation[PTAnnotPageNumberKey]).intValue;
+                if (annotationId.length > 0) {
+                    PTAnnot *annot = [PTPluginUtils findAnnotWithUniqueID:annotationId onPageNumber:pageNumber documentViewController:docVC];
```

Comment:
This could potentially be very slow, it's n^2 for the number of annotations. Would be nicer to write this in a way that took a vector of annotIDs and looped through them once and returned them as an array.

Not sure this is worth doing now though, if there are a sane number of annotations it should still be OK.